apps/src/sites/studio/pages/googleBlocklyWrapper.js

```diff
@@ -155,7 +156,30 @@ function initializeBlocklyWrapper(blocklyInstance) {
       RUN_BUTTON_CLICKED: 'runButtonClicked'
     },
     onMainBlockSpaceCreated: () => {}, // TODO
-    createReadOnlyBlockSpace: () => {} // TODO
+    createReadOnlyBlockSpace: (container, xml, options) => {
+      const workspace = new Blockly.WorkspaceSvg({
+        readOnly: true,
+        theme: CdoTheme
+      });
+      const svg = Blockly.utils.dom.createSvgElement(
+        'svg',
+        {
+          xmlns: 'http://www.w3.org/2000/svg',
+          'xmlns:html': 'http://www.w3.org/1999/xhtml',
+          'xmlns:xlink': 'http://www.w3.org/1999/xlink',
+          version: '1.1',
+          class: 'geras-renderer modern-theme readOnlyBlockSpace'
+        },
+        null
+      );
+      container.appendChild(svg);
+      svg.appendChild(workspace.createDom());
+      Blockly.Xml.domToBlockSpace(workspace, xml);
+      const bbox = svg.getBBox();
+      svg.setAttribute('height', bbox.y + bbox.height + bbox.y);
+      svg.setAttribute('width', bbox.x + bbox.width + bbox.x);
```

Comment:
Out of curiosity, what is happening here?  Is `bbox.y`/`bbox.x` some kind of padding, which is why it's appearing twice?

Reply:
this is just to shrink the svg to the size of the block. I got this solution from here https://stackoverflow.com/questions/50813950/how-do-i-make-an-svg-size-to-fit-its-content, and didn't notice that x and y were being added twice. After looking at it more carefully (and referencing https://www.w3.org/Graphics/SVG/IG/resources/svgprimer.html#getBBox), I think we should only need to add the x and y offsets once.